upup/pkg/fi/cloudup/azuretasks/loadbalancer.go

```diff
@@ -0,0 +1,233 @@
+	}
+	feConfig := feConfigs[0]
+	subnet := feConfig.FrontendIPConfigurationPropertiesFormat.Subnet
+
+	return &LoadBalancer{
+		Name:      lb.Name,
+		Lifecycle: lb.Lifecycle,
+		ResourceGroup: &ResourceGroup{
+			Name: lb.ResourceGroup.Name,
+		},
+		Subnet: &Subnet{
+			Name: subnet.Name,
+		},
+		RequirePublicIP: to.BoolPtr(feConfig.FrontendIPConfigurationPropertiesFormat.PublicIPAddress != nil),
+		Tags:            found.Tags,
+	}, nil
+}
+
+// Run implements fi.Task.Run.
+func (lb *LoadBalancer) Run(c *fi.Context) error {
+	c.Cloud.(azure.AzureCloud).AddClusterTags(lb.Tags)
+	return fi.DefaultDeltaRunMethod(lb, c)
+}
+
+// CheckChanges returns an error if a change is not allowed.
+func (*LoadBalancer) CheckChanges(a, e, changes *LoadBalancer) error {
+	if a == nil {
+		// Check if required fields are set when a new resource is created.
+		if e.Name == nil {
+			return fi.RequiredField("Name")
+		}
+		return nil
+	}
+
+	// Check if unchanegable fields won't be changed.
+	if changes.Name != nil {
+		return fi.CannotChangeField("Name")
+	}
+	return nil
+}
+
+// RenderAzure creates or updates a Loadbalancer.
+func (*LoadBalancer) RenderAzure(t *azure.AzureAPITarget, a, e, changes *LoadBalancer) error {
+	if a == nil {
+		klog.Infof("Creating a new Loadbalancer with name: %s", fi.StringValue(e.Name))
+	} else {
+		klog.Infof("Updating a Loadbalancer with name: %s", fi.StringValue(e.Name))
+	}
+
+	idPrefix := fmt.Sprintf("subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network", t.Cloud.SubscriptionID(), *e.ResourceGroup.Name)
+	feConfigProperties := &network.FrontendIPConfigurationPropertiesFormat{}
+	if *e.RequirePublicIP {
+		feConfigProperties.PublicIPAddress = &network.PublicIPAddress{
+			ID: to.StringPtr(fmt.Sprintf("/%s/publicIPAddresses/%s", idPrefix, *e.PublicIPName)),
+		}
+	} else {
+		feConfigProperties.PrivateIPAllocationMethod = network.Dynamic
+		feConfigProperties.Subnet = &network.Subnet{
+			ID: to.StringPtr(fmt.Sprintf("/%s/virtualNetworks/%s/subnets/%s", idPrefix, *e.Subnet.VirtualNetwork.Name, *e.Subnet.Name)),
+		}
+	}
+	lb := network.LoadBalancer{
+		Location: to.StringPtr(t.Cloud.Region()),
+		Sku: &network.LoadBalancerSku{
+			Name: network.LoadBalancerSkuNameBasic,
+		},
+		LoadBalancerPropertiesFormat: &network.LoadBalancerPropertiesFormat{
+			FrontendIPConfigurations: &[]network.FrontendIPConfiguration{
+				{
+					Name:                                    to.StringPtr("LoadBalancerFrontEnd"),
+					FrontendIPConfigurationPropertiesFormat: feConfigProperties,
+				},
+			},
+			BackendAddressPools: &[]network.BackendAddressPool{
+				{
+					Name: to.StringPtr("LoadBalancerBackEnd"),
+				},
+			},
+			Probes: &[]network.Probe{
+				{
+					Name: to.StringPtr("Health-TCP-80"),
+					ProbePropertiesFormat: &network.ProbePropertiesFormat{
+						Protocol:          network.ProbeProtocolTCP,
+						Port:              to.Int32Ptr(80),
```

Comment:
Question: Do we need to have a probe for port 80? Or just 443 will be sufficient?